src/Components/animation/Animation.js

```diff
@@ -137,377 +69,265 @@ class Animation extends React.Component {
-          pixelPosition = this._positionPercentToPixel(
-            nextPosition[substepCount - 1][0],
-            nextPosition[substepCount - 1][1],
-          );
+  /** Instantly move to a step */
+  moveToStep = stepId => {
+    this.setState(
+      {
+        animationPlaying: true,
+      },
+      () => {
+        var stepId2 = Math.max(0, Math.round(stepId));
+        Animated.timing(this.currentStepAV, {
+          toValue: stepId2,
+          duration: 0,
+          easing: Easing.linear,
+          key: 0,
+        }).start(() => this.setState({ animationPlaying: false, currentStep: stepId2 }));
+      },
+    );
+  };
 
-          /* Get the corresponding animation */
-          anim = currentDisplayedElement.getAnimation(pixelPosition[0], pixelPosition[1], this.state.stepLength);
+  /** Animate a step (or just move instantly to it is the first step) */
+  playStep = stepId => {
+    this.setState(
+      {
+        animationPlaying: true,
+      },
+      () => {
+        stepId = Math.max(0, Math.round(stepId));
+
+        var sequence = [];
+
+        /* Move instantly to the previous step */
+        sequence.push(
+          Animated.timing(this.currentStepAV, {
+            toValue: Math.max(0, stepId - 1),
+            duration: 0,
+            easing: Easing.linear,
+            key: 0,
+          }),
+        );
 
-          displayedElementStepAnimation.push(anim);
+        /* If we do not move to the first step, animate the move from the previous step to the current step */
+        if (stepId > 0) {
+          sequence.push(
+            Animated.timing(this.currentStepAV, {
+              toValue: stepId,
+              duration: this.state.stepLength,
+              easing: Easing.linear,
+              key: 1,
+            }),
+          );
         }
 
-        stepAnimation.push(Animated.sequence(displayedElementStepAnimation));
-      }
-    }
-
-    return stepAnimation;
+        Animated.sequence(sequence).start(() => this.setState({ animationPlaying: false, currentStep: stepId }));
+      },
+    );
   };
 
-  _display(item) {
-    if (item !== undefined && item !== null) return item.render();
-    else return <View />;
-  }
-
   render() {
-    this.cutsArray = [];
+    //<MaterialCommunityIcons name="chevron-double-left" color={theme.COLOR_PRIMARY} size={26} />
 
     return (
       <View style={[styles.mainContainer, { height: this.animationHeight }, { width: this.animationWidth }]}>
-        {this.props.editable && this.cuts !== undefined && this.cuts !== null ? (
-          <DisplayedCuts step={this.state.currentStep} drillCuts={this.cuts} />
+        {this.props.editable && !this.state.animationPlaying ? (
+          <DisplayedCuts
+            step={this.state.currentStep}
+            positionPercentToPixel={this._positionPercentToPixel}
+            animation={this.props.animation}
+            onMoveEnd={this.props.onCutMove}
+          />
         ) : (
```

Comment:
You can remove the undefined if you write the condition this way :

```jsx
        {this.props.editable && !this.state.animationPlaying && (
          <DisplayedCuts
            step={this.state.currentStep}
            positionPercentToPixel={this._positionPercentToPixel}
            animation={this.props.animation}
            onMoveEnd={this.props.onCutMove}
          />
        )}
```

It's closer to the "react" way of writing conditions within render

Reply:
Done